File: java/src/main/java/dev/jeffrosenberg/adventofcode/day01_2017/Captcha.java
package dev.jeffrosenberg.adventofcode.day01_2017;

public class Captcha {
  public static int getCaptcha(String input) {
    return getCaptcha(input, 1);
  }

  public static int getCaptcha(String input, int step) {
    int result = 0;
    char[] inputChars = input.toCharArray();

    for(int i = 0; i < inputChars.length; i++) {
      int j = ((i + step) % inputChars.length);
      if (j >= inputChars.length) { j = 0; }
      if(inputChars[i] == inputChars[j]) {
        result += Character.getNumericValue(inputChars[i]);
      }
    }

    return result;
  }
}

File: csharp/AdventOfCode/2017/201701/Captcha.cs
using System;

namespace AdventOfCode
{
    public class Captcha
    {
        public static int GetCaptcha(string input, int step = 1)
        {
            int result = 0;
            char[] inputChars = input.ToCharArray();
            for(int i = 0; i < inputChars.Length; i++) 
            {
                int j = ((i + step) % inputChars.Length);
                if(inputChars[i] == inputChars[j]) 
                {
                    result += int.Parse(inputChars[i].ToString());
                }
            }

            return result;
        }
    }
}
